client/src/components/RelatedProducts/Cards/AllCards.jsx

```diff
@@ -1,41 +1,43 @@
 import React from 'react';
 import styled from 'styled-components';
-import { Individualcard, IndCard } from './Individualcard';
-import ProductImg from './ProductImg';
-import sampleRelatedData from './sampleRelatedData';
-import sampleRelatedThumbnails from './sampleRelatedThumbnails';
+import { Individualcard } from './Individualcard';
+import { useProd } from './ProdContext';
+import { useThumbnail } from './thumbnailContext';
+
+function AllCards() {
+  const relatedProds = useProd();
+  const thumbnail = useThumbnail();
+
+  const zippedArray = [];
+  if (!thumbnail.data) {
+    console.log('loading thumbnail');
+  } if (!relatedProds.data) {
+    console.log('loading related prods');
+  } else {
+    relatedProds.data.map((item, i) => {
+      zippedArray.push([item.data, thumbnail.data[i].data.results[0].photos[0].thumbnail_url]);
+    });
+  }
+
+  if (!relatedProds.data) {
+    console.log('loading Prods');
+  } else {
+    return (
+
+      <Layout>
+        {zippedArray.map((item, key) => <div><Individualcard product={item} key={key} /></div>)}
```

Comment:
I think the reason this is still throwing an error is because the `div` element needs the `key`, not the `IndividualCard`. Although it looks like the `div` isn't styled - do you need the `div` at all here?

Reply:
Removed random div